socorro/signature/rules.py

```diff
@@ -507,20 +527,23 @@ def action(self, crash_data, result):
                 signature_list = []
 
         except (KeyError, IndexError) as exc:
-            signature_notes.append('No crashing frames found because of %s' % exc)
+            result.note('No crashing frames found because of %s', exc)
             signature_list = []
 
-        signature, signature_notes = self.c_signature_tool.generate(
+        signature, notes, debug_notes = self.c_signature_tool.generate(
             signature_list,
             crash_data.get('hang_type'),
             crash_data.get('crashing_thread')
         )
 
         if signature_list:
-            result['proto_signature'] = ' | '.join(signature_list)
+            result.extra['proto_signature'] = ' | '.join(signature_list)
```

Comment:
Should double-check there's a test that makes sure the `proto_signature` makes it into the processed crash.